src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,204 @@
+        List<Product> productList = new ArrayList<>();
+        do {
+            productList.add(parseProduct(sqlResult));
+        } while (sqlResult.next());
+        return new Order(id, userId, productList);
+    }
+
+    private Product parseProduct(ResultSet sqlResult) throws SQLException {
+        Long productId = sqlResult.getLong("product_id");
+        String name = sqlResult.getString("name");
+        Double price = sqlResult.getDouble("price");
+        return new Product(productId, name, price);
+    }
+
+    @Override
+    public Order update(Order order) {
+        String query = "DELETE FROM orders_products"
+                        + " WHERE orders_id = ?";
+
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, order.getId());
+            preparedStatement.executeUpdate();
+            insertProducts(order, connection);
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't update order for user = " + order.getUserId(), e);
+        }
+        return order;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE orders"
+                        + " SET deleted = true"
+                        + " WHERE order_id = ? and deleted = false";
+
+        String query2 = "DELETE FROM orders_products"
+                        + " WHERE orders_id = ?";
+
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement prepareStatement = connection.prepareStatement(query);
+            prepareStatement.setLong(1, id);
+            prepareStatement.executeUpdate();
+            prepareStatement = connection.prepareStatement(query2);
+            prepareStatement.setLong(1, id);
+            prepareStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Order with id =" + id
+                    + " has not been deleted", e);
+        }
+        return true;
+    }
+
+    @Override
+    public List<Order> getAll() {
+        String query = "SELECT * FROM orders AS o"
+                        + " JOIN orders_products AS op ON o.order_id = op.orders_id"
+                        + " JOIN products AS p ON p.product_id = op.products_id"
+                        + " WHERE o.deleted = false ";
+
+        List<Order> orderList = new ArrayList<>();
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            ResultSet sqlResult = preparedStatement.executeQuery();
+            while (sqlResult.next()) {
+                Long id = sqlResult.getLong("order_id");
+                Long userId = sqlResult.getLong("user_id");
+                Order order = new Order(id, userId);
```

Comment:
Optional = you can skip this line and on the next line - "... .add(new Order (...))

Reply:
Ok. Done